R/Widget_GroupOverview.R

```diff
@@ -53,12 +53,24 @@ Widget_GroupOverview <- function(
   # set strGroupLevel if NULL and dfMetrics is not NULL
   if (is.null(strGroupLevel) && !is.null(dfMetrics)) {
     strGroupLevel <- unique(dfMetrics$GroupLevel)
-  } else if (!is.null(strGroupLevel)) {
-    strGroupLevel <- strGroupLevel
   } else {
     stop("One of strGroupLevel or dfMetrics must be provided to create group-level output.")
   }
 
+  # subset dfResults on latest snapshot date
+  if (length(unique(dfResults$SnapshotDate)) > 1) {
+    latest_snapshot_date <- max(dfResults$SnapshotDate)
+
+    cli::cli_alert_warning(
+      "[ dfResults ] contains multiple snapshot dates and will be subset on the latest snapshot date: {latest_snapshot_date}."
+    )
+
+    dfResults <- dfResults %>%
+      dplyr::filter(
+        .data$SnapshotDate == latest_snapshot_date
+      )
+  }
+
```

Comment:
I made a convenience wrapper, `filter_by_latest_SnapshotDate()`, to handle & standardize this. Adding the warning when we apply that would probably be generally useful to add in `filter_by_latest_SnapshotDate()`, though (but only when `length(unique(dfResults$SnapshotDate > 1)`).

Reply:
Anywhere we're filtering on latest snapshot date has been updated to use this function.